### PR TITLE
Use MacOS-13 in CI

### PR DESCRIPTION
**Pull request by wtgodbe**

### CODE REVIEW

File: src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/DefaultHttpClient.java [hunk 1]
```diff
@@ -116,9 +116,9 @@ public Single<HttpResponse> send(HttpRequest httpRequest, ByteBuffer bodyContent
             case "POST":
                 RequestBody body;
                 if (bodyContent != null) {
-                    body = RequestBody.create(MediaType.parse("text/plain"), ByteString.of(bodyContent));
+                    body = RequestBody.Companion.create(ByteString.of(bodyContent), MediaType.parse("text/plain"));
```
Comment (BrennanConroy) [inline src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/DefaultHttpClient.java:119]: https://github.com/square/okhttp/blob/911c5bf5b27e8378f209861a16a9da4cc54cfff7/okhttp/src/main/java/okhttp3/RequestBody.kt#L168

File: src/SignalR/clients/java/signalr/core/build.gradle [hunk 2]
```diff
@@ -1,60 +1,67 @@
...
             }
         }
-    }.writeTo("${buildDir}/libs/signalr-${project.version}.pom")
+    }
 }
 
-task createPackage(dependsOn: [jar,sourceJar,javadocJar,generatePOM])
+tasks.withType(GenerateMavenPom).all {
+    def matcher = name =~ /generatePomFileFor(\w+)Publication/
+    def publicationName = matcher[0][1]
```
Comment (BrennanConroy) [inline src/SignalR/clients/java/signalr/core/build.gradle:62]:
```suggestion
```

File: src/SignalR/clients/java/signalr/messagepack/build.gradle [hunk 3]
```diff
@@ -1,56 +1,64 @@
...
             }
         }
-    }.writeTo("${buildDir}/libs/signalr-messagepack-${project.version}.pom")
+    }
 }
 
-task createPackage(dependsOn: [jar,sourceJar,javadocJar,generatePOM])
+tasks.withType(GenerateMavenPom).all {
+    def matcher = name =~ /generatePomFileFor(\w+)Publication/
+    def publicationName = matcher[0][1]
```
Comment (BrennanConroy) [inline src/SignalR/clients/java/signalr/messagepack/build.gradle:62]:
```suggestion
```

**Review by BrennanConroy — APPROVED** [review]

File: global.json [hunk 4]
```diff
@@ -24,7 +24,7 @@
     "xcopy-msbuild": "17.1.0"
   },
   "native-tools": {
-    "jdk": "11"
+    "jdk": "11.0.24"
```
Comment (ViktorHofer) [inline global.json:27]: @wtgodbe this change broke the entire VMR now that images got updated to a newer version of the jdk. This already happened once in the past which is why we made this change: https://github.com/dotnet/aspnetcore/commit/4a3afe3b23e062ca120c08a2761b3ea7775a03f9

Can you please describe why the minor and patch versions were necessary to be added here?

cc @mmitche

Comment (ViktorHofer) [reply global.json:27]: Arcade offers scripts to install native tools which support defining just a major version or even "latest": https://github.com/dotnet/aspnetcore/blob/main/eng/common/init-tools-native.ps1

Any reason why you can't use those? The problem with the hardcoded version is that it breaks whenever the CI images get updated. I.e. right now all aspnetcore builds should be broken. See https://github.com/dotnet/sdk/pull/45822

Comment (wtgodbe) [reply global.json:27]: I don't see any reason why we can't - I'd actually prefer it. Do we just need to add a pre-step in CI that runs the script?

Comment (wtgodbe) [reply global.json:27]: We used to install the JDK manually ourselves w/ https://github.com/dotnet/aspnetcore/blob/main/eng/scripts/InstallJdk.ps1, but stopped with https://github.com/dotnet/aspnetcore/pull/54421. I was under the impression we had to use whatever was pre-installed on the build machine/in the VMR tool cache

Comment (ViktorHofer) [reply global.json:27]: Let's move that discussion to https://github.com/dotnet/source-build/issues/4816

**Pull request merged**